docs/proposed/config-split.md

```diff
@@ -0,0 +1,450 @@
+like platform or system ones are moved to `(tf_psa_)crypto_config.h`. That way
+they are available in both repositories (as Mbed TLS includes
+`tf_psa_crypto_config.h`) without duplication. Later, we may duplicate or
+create aliases for some of them to align with the naming conventions of the
+repositories.
+
+The layout of options into sections in `mbedtls_config.h` does not suit
+TF-PSA-Crypto well thus the configuration options `tf_psa_crypto_config.h` are
+organized into different sections (see below).
+
+## Configuration files and `config.py`
+
+Each repository contains a `config.py` script to create and modify
+configurations.
+
+In Mbed TLS, `config.py` handles both `mbedtls_config.h` and
+`tf_psa_crypto_config.h`. It can set or unset TLS, x509 and cryptographic
+configuration options without having to specify the configuration file the
+options belong to. Commands like full and baremetal affect both configuration
+files.
+
+In TF-PSA-Crypto, `config.py` addresses only `tf_psa_crypto_config.h`.
+
+## Sections in `tf_psa_crypto_config.h`
+
+The `tf_psa_crypto_config.h` configuration file is organized into eight
+sections.
+
+The pre-split `mbedtls_config.h` configuration file contains configuration
+options that apply to the whole code base (TLS, x509, crypto and tests) mostly
+related to the platform abstraction layer and testing. In
+`tf_psa_crypto_config.h` these configurations options are organized into two
+sections, one for the platform abstraction layer options and one for the others,
+respectively named "Platform abstraction layer" and
+"General and test configuration options".
+
+Then, the "Cryptographic mechanism selection (PSA API)" section is the
+equivalent of the pre-split `crypto_config.h` configuration file containing the
+PSA_WANT_ prefixed macros.
+
+The following section named "Cryptographic mechanism selection (extended API)"
+contains the configuration options for the cryptography mechanisms that are not
+yet part of the PSA cryptography API (like LMS or PK).
+
+It is followed by the "Data format support" section that contains configuration
+options of utilities related to various data formats (like base64 or ASN1 APIs).
+These utilities aim to facilitate the usage of the PSA cryptography API in other
+cryptography projects.
+
+Compared to Mbed TLS, the cryptography code in TF-PSA-Crypto is not located
+in a single directory but split between the PSA core (core directory) and the
+PSA builtin drivers (drivers/builtin/src directory). This is reflected in
+`tf_psa_crypto_config.h` with two sections respectively named "PSA core" and
+"Builtin drivers".
+
+Finally, the last section named "Legacy cryptography" contains the configuration
+options that will eventually be removed as duplicates of PSA_WANT_\* and
+MBEDTLS_PSA_ACCEL_\* configuration options.
+
+By contrast to `mbedtls_config.h`, `tf_psa_crypto_config.h` does not contain a
+section like the "Module configuration options" one containing non boolean
```

Comment:
Why would `mbedtls_config.h` keep a separate section for "Module configuration options"? Either it's a problem with our tooling and both projects need to keep non-boolean options in separate sections, or the tooling is fine and I don't see why we wouldn't merge "Module configuration options" with "Mbed TLS feature support".

Reply:
After discussion and experiment, it _is_ a problem with our tooling, but one that it's easier to solve than work around.

https://github.com/Mbed-TLS/mbedtls/pull/9604 removes the reliance of `config.py` on which section a setting is in.

Reply:
No section dedicated to non boolean option anymore, both in tf_psa_crypto_config.h and mbedtls_config.h.